x-pack/plugins/lens/public/editor_frame_service/error_helper.ts

```diff
@@ -13,6 +13,7 @@ interface ElasticsearchErrorClause {
   type: string;
   reason: string;
   caused_by?: ElasticsearchErrorClause;
+  failed_shards?: Array<{ reason: ElasticsearchErrorClause & { script?: string; lang?: string } }>;
```

Comment:
It looks like this isn't typed as part of the `@elastic/elasticsearch` native types that we offer. Maybe we can use `data/common/search/search_source/types.ts` instead of writing our own here? It would also be nice if we could get error types from the elasticsearch-js client. cc @delvedor

Reply:
Currently, we are shipping a type with the generic structure of Elasticsearch errors, there is an issue (https://github.com/elastic/elastic-client-generator/issues/340) about providing more advanced types, but it's not our top priority right now.